lib/libthr/thread/thr_stack.c

```diff
@@ -126,10 +126,7 @@ static char *last_stack = NULL;
 static inline size_t
 round_up(size_t size)
 {
-	if (size % _thr_page_size != 0)
-		size = ((size / _thr_page_size) + 1) *
-		    _thr_page_size;
-	return size;
+	return (roundup2(size, _thr_page_size - 1));
```

Comment:
```suggestion
	return (roundup2(size, _thr_page_size));
```

I'd merged this, but Doug Moore correctly pointed out that this is wrong.

Reply:
You're right.  Too many versions of code going around and I stopped thinking about each new one.  It will usually work, but the `- 1` is wrong.

Reply:
I pushed a fix to the branch associated with this pull request.